### PR TITLE
__int128 <-> _Decimal128 DPD Conversion Support

### PR DESCRIPTION
**Pull request by pmur**
BID _Decimal128 -> __int128 support is still TODO.

### CODE REVIEW

**Comment by pmur** [comment]
I reworked the conversion routines to support dd and df support, and removed a bit of duplicate code.

BID is failing tests unexpectedly, some comparisons seem to be getting evaluated wrong. (i.e -1 > 0 returning true). I need to look into that.

File: sysdeps/dpd/Makefile [hunk 1]
```diff
@@ -1 +1 @@
-libdfp_files += dpd-private binpowof10 decpowof2
+libdfp_files += dpd-private binpowof10 decpowof2 ticonstants fixtdti fixunstdti
```
Comment (tuliom) [inline sysdeps/dpd/Makefile]: I can't build these files on ppc32.

**Comment by tuliom** [comment]
This PR is breaking the check-localplt.out on ppc64.

**Comment by pmur** [comment]
So, TImode probably won't ever work on a 32 bit platform. The build system needed a bit of monkeying as this causes us to support another ABI without TImode.

Unfortunately, this makes the tedious task of maintaining the ABI check 2x as time consuming. 

I'm not going to add a configuration option to control this, as it will become increasingly complicated to maintain as float128 will also depend on this.

**Comment by pmur** [comment]
Fixed a typo symbol-hack.h causing unsdddi to get linked to unsddti, causing a test failure with soft-dfp.

**Comment by pmur** [comment]
Thanks.

I agree, the ctz/clz bug *should* have been picked up by the test suite. However, the test suite does not do much (any?) testing of proper exception handling.

**Comment by pmur** [comment]
Fixed the comment and merged via a rebase.

**Pull request closed without merging**